### PR TITLE
Deprecate envoy.reloadable_features.postpone_h3_client_connect_to_nex…

### PR DESCRIPTION
**Pull request by sgovinda-ub**
This change is to deprecate the runtime guard,
envoy_reloadable_features_postpone_h3_client_connect_to_next_loop .

Signed-off-by: Shyam Govindaraj [shyirtt@gmail.com](mailto:shyirtt@gmail.com)

Commit Message: deprecate runtime guard(envoy_reloadable_features_postpone_h3_client_connect_to_next_loop)
Risk Level: Low
Testing: Done
Docs Changes: N/A
Release Notes: N/A
Fixes: https://github.com/envoyproxy/envoy/issues/23600

### CODE REVIEW

**Comment by sgovinda-ub** [comment]
@danzh2010  Please review

**Comment by danzh2010** [comment]
Can you makes the CIs happy?

File: test/common/http/http3/conn_pool_test.cc [hunk 1]
```diff
@@ -195,11 +189,6 @@ TEST_F(Http3ConnPoolImplTest, CreationAndNewStream) {
 }
 
 TEST_F(Http3ConnPoolImplTest, NewAndCancelStreamBeforeConnect) {
-  if (!Runtime::runtimeFeatureEnabled(
```
Comment (danzh2010) [inline test/common/http/http3/conn_pool_test.cc:198]: This test is testing the false case, and should be removed.

Comment (danzh2010) [reply test/common/http/http3/conn_pool_test.cc:198]: sorry, I commented on the wrong one.

File: test/common/http/conn_pool_grid_test.cc [hunk 2]
```diff
@@ -956,9 +955,6 @@ TEST_F(ConnectivityGridTest, RealGrid) {
 }
 
 TEST_F(ConnectivityGridTest, ConnectionCloseDuringCreation) {
-  TestScopedRuntime scoped_runtime;
-  scoped_runtime.mergeValues(
-      {{"envoy.reloadable_features.postpone_h3_client_connect_to_next_loop", "false"}});
```
Comment (danzh2010) [inline test/common/http/conn_pool_grid_test.cc:961]: This test is testing the false case, and should be removed.

Comment (sgovinda-ub) [reply test/common/http/conn_pool_grid_test.cc:961]: Done

File: source/common/http/http3/conn_pool.cc [hunk 3]
```diff
@@ -188,15 +186,9 @@ allocateConnPool(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_
...
-        CodecClientPtr codec =
-            Runtime::runtimeFeatureEnabled(
-                "envoy.reloadable_features.postpone_h3_client_connect_to_next_loop")
-                ? std::make_unique<NoConnectCodecClientProd>(
-                      CodecType::HTTP3, std::move(data.connection_), data.host_description_,
-                      pool->dispatcher(), pool->randomGenerator(), pool->transportSocketOptions())
-                : std::make_unique<CodecClientProd>(
-                      CodecType::HTTP3, std::move(data.connection_), data.host_description_,
-                      pool->dispatcher(), pool->randomGenerator(), pool->transportSocketOptions());
+        CodecClientPtr codec = std::make_unique<NoConnectCodecClientProd>(
```
[the remark this answers is not in the document]
Comment (sgovinda-ub) [reply source/common/http/http3/conn_pool.cc:189]: @danzh2010  Can I remove the if check(line no:177 to 182) in the same PR?

[the remark this answers is not in the document]
Comment (danzh2010) [reply source/common/http/http3/conn_pool.cc:189]: Yes, since it lowers the coverage.

[the remark this answers is not in the document]
Comment (sgovinda-ub) [reply source/common/http/http3/conn_pool.cc:181]: Ack

**Comment by sgovinda-ub** [comment]
@danzh2010  Can you please review?

**Review by danzh2010 — APPROVED** [review]
Thanks for making the change!

**Comment by sgovinda-ub** [comment]
@danzh2010 can we merge this ?

File: source/common/http/http3/conn_pool.cc [hunk 4]
```diff
@@ -173,30 +171,16 @@ allocateConnPool(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_
               "Failed to create Http/3 client. Failed to create quic network connection.");
           return nullptr;
         }
-        // Store a handle to connection as it will be moved during client construction.
-        Network::Connection& connection = *data.connection_;
         auto client = std::make_unique<ActiveClient>(*pool, data);
-        if (connection.state() == Network::Connection::State::Closed) {
```
Comment (alyssawilk) [inline source/common/http/http3/conn_pool.cc:179]: @danzh2010 in future it'd be nice to have an ASSERT(runtime_guard_false) so it's clear the path is safe to delete.

**Pull request merged**